### DISCUSSION TITLE
⏳ 🔗 Order Loop Grid by Custom Field

### DISCUSSION THREAD

**Discussion by GwendleWeb** [elementor/elementor#22156] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Currently you can only order the loop grid using 6 set options (Date, Title, Menu Order, Last Modified, Comment Count, Random) this is ok for a standard blog but if you're using the grid to display custom post times with custom fields you are really limited.  

Eg I am creating a dynamic gig list for a band,  I have a custom post type for events which includes a custom field for the date of event.  I need to display the gigs in a list ordered by this custom date field.   As the gigs can be created in advance and published at a random date that is different to the date of the gig you cannot use the post date to correctly order the list.  Menu order is not available for posts.   A work around is to carefully set the title of each gig post to reflect the event date however as the event date is already set up in the post it would be much more convenient to pull that field through for use in the query. 

There are other situations where being able to order posts by custom fields would be really useful such as a real estate website if you wanted to be able to order properties by price, a walking website where you want to order the walks by length etc

### Describe the solution you'd like

The loop grid query allows you to order by a custom field

### Additional context

 There is a plug in called post types order which you can install that lets you drag and drop to order your posts but the fewer plugins you need the better and this would be really time consuming if you have a massive list of posts and need to change the order.

**Comment by tinymachine** [comment]
Fully agreed with @GwendleWeb. There's another plugin called [Advanced Post Queries][apq] that's supposed to let you sort loop grids by [Advanced Custom Fields][acf], which would address this need. But as of the latest version 1.1.1 released around May 17, it's crashing my Elementor-hosted site (running WordPress 6.2.2 and Elementor & Elementor Pro 3.14.0-cloud2). 

I filed a [support ticket][st] two weeks ago, but no response yet, which highlights why this functionality should be built into Elementor and not handled by a plugin.

[apq]: https://wordpress.org/plugins/advanced-post-queries/
[acf]: https://wordpress.org/plugins/advanced-custom-fields/
[st]: https://wordpress.org/support/topic/activating-the-plulgin-immediately-crashed-my-site/

**Reply by alsage13** [reply]
I've run into the same thing with this plugin, unfortunately. I find it very complicated to use and there doesn't seem to be any support now.

**Comment by micmiller-apptio** [comment]
I end up writing custom query filters in PHP to sort posts by different ACF values. I expect to write filters for more complex query rules, but it's a pain to add code for something as simple as sorting on a given field.

**Comment by JeremyCarlsten** [comment]
This would be awesome!

**Reply by JeremyCarlsten** [reply]
I was able to get around this using a plugin https://wordpress.org/plugins/advanced-post-queries/. I don't like adding a new plugin for something that should be baked in. If a post_type has the field we should be able to use it is my opinion. 

But for now that works, if not a little quirky.

**Comment by nadworks** [comment]
Being able to easily sort a loop grid by ACFs - i.e. the date picker field, combined with the options of filtering past/upcoming dates, and filtering by year would make the new loop grid incredibly powerful and finally move on from the last remnants of WP's "blogging" legacy.

Same as @GwendleWeb, we are building websites for musicians, bands and venues who have the need to display custom concert calendars that sort by performance dates, not post dates. The need for custom queries makes this endeavour pretty clunky when it could be much easier.

**Comment by alsage13** [comment]
This is HUGE for me. I have a site right now that needs this feature in about 6 different places, and since I'm not comfortable with writing my own queries, I'm having to pay a developer to do so. What I'm looking for is twofold:

- Filter by a specific ACF field (usually by a certain response from a select box or radio button)
- Filter by an ACF date field, specify upcoming and past posts (similarly to the others, I am working on a site with a lot of events and I'd like to be able to easily show upcoming events, and those that have already passed.)

I'm sure I can come up with additional uses, but these two are enormous for me!

**Comment by lllopo** [comment]
+100 ... Currently the loop queries are very, very limited. Adding custom fields support for both filtering and sorting will really 'unlock' this feature and make it a really powerful tool.

**Comment by DownHouse** [comment]
This code sort posts by acf field in the query with "query_id" id.

```
function my_query_by_different_order( $query ) {
    $query->set( 'orderby', 'meta_value' );
    $query->set( 'meta_key', 'acf_field_name' );
}
add_action( 'elementor/query/query_id', 'my_query_by_different_order' );
```

Work fine with date field.

https://developers.elementor.com/docs/hooks/custom-query-filter/

**Reply by klavsveg99** [reply]
Loop Grids do not offer a Query ID input at the moment. To do this with the Posts widget, you need Ele Custom Skin Pro https://dudaster.com/ecs-pro/

**Reply by DownHouse** [reply]
This input field has always been there. It is definitely there in version 3.8 and more.

**Reply by klavsveg99** [reply]
You're right, that's because I had Template Type set to "Product" instead of "Posts" under the Layout tab for the Loop Grid block

**Reply by RonR-WebDesign** [reply]
DownHouse, this works perfectly and is a simple solution.  THANK YOU!

**Reply by lensbreak** [reply]
Why the heck is this not possible in  if the template Type set to "Product" instead of "Posts"?

**Reply by codewizard13** [reply]
@lensbreak I'd like to know the reason behind this also. This is curious @elementorbot

**Reply by awebguy99** [reply]
> This code sort posts by acf field in the query with "query_id" id.
> 
> ```
> function my_query_by_different_order( $query ) {
>     $query->set( 'orderby', 'meta_value' );
>     $query->set( 'meta_key', 'acf_field_name' );
> }
> add_action( 'elementor/query/query_id', 'my_query_by_different_order' );
> ```
> 
> Work fine with date field.
> 
> https://developers.elementor.com/docs/hooks/custom-query-filter/

I'm not really good at coding. Do I just add this as a PHP snippet and replace **acf_field_name** with the actual ACF field name?

function my_query_by_different_order( $query ) {
    $query->set( 'orderby', 'meta_value' );
    $query->set( 'meta_key', '**acf_field_name**' );
}
add_action( 'elementor/query/query_id', 'my_query_by_different_order' );

**Reply by DownHouse** [reply]
Add this code to functions.php.
Replace "acf_field_name" with the your ACF field name.
Replace "query_id" with this ID in loop grid:
![Скриншот 28-05-2024 015704](https://github.com/elementor/elementor/assets/5120629/1aef487e-c492-4eaa-94b6-a45c6fbe0ab8)

**Reply by awebguy99** [reply]
Thanks @DownHouse !! :)

**Comment by Asher-ly** [comment]
I often am comparing the perks of builders and this feature is what makes Webflow CMS win- the ability to specifically define sorting, as well as the ability to hide or show something based off of a custom field being filled in!

**Reply by lensbreak** [reply]
good luck with the vendor lock-in, now you are bound on their pricing and policies.

**Comment by kinokeo** [comment]
I would also find this super useful!!

**Comment by ppwebox** [comment]
Absolutely a needed feature! But even much more so is to have a front-end option to let visitors choose the sorting option! Be it via dropdown menu or radio buttons. It's mystery to me why the devs left out such an important feature for loop grids.

**Comment by csqshiqi** [comment]
Hi, i too feel that this feature is necessary. Unlike a blog, many websites using elementor are trying to showcase products or services that we provide. I faced this issue when using the Loop Grid object on my page, we would like to feature certain products at the top and there was no way i can define the order of products aside from these options:
![image](https://github.com/elementor/elementor/assets/147570956/57f50fcb-dd53-44f7-b6b3-e7703beccdfe)

Right now i'm forced to edit the title to include numbers at the beginning just so i can force the order to be sorted accordingly.

**Comment by pumadog** [comment]
I'm currently using the plugin "Advanced Queries" for this, but yes, that should be an Elementor feature.

**Comment by lensbreak** [comment]
PLEASE make a front-end option to let visitors choose the sorting option, e.g. a dropdown menu!

**Comment by koonda** [comment]
Can you guys focus on what you have to enhance these functionalities instead of rolling out in new updates half finished features? This is driving me crazy... You always promise to add improvements later on but never deliver.

**Reply by awebguy99** [reply]
> Can you guys focus on what you have to enhance these functionalities instead of rolling out in new updates half finished features? This is driving me crazy... You always promise to add improvements later on but never deliver.

Yeah, it's the most frustrating thing about Elementor. Time and time again, they release something with about 80 to 90% of the features it needs. Then they move onto the next new thing.

**Comment by ppwebox** [comment]
True to an extent also with the Elementor widgets. Again and again I'm bewildered about how half-baked these were put into production. Many widgets also require custom CSS out of the box to look half-way decent (i.e., professional).

**Comment by izzymg** [comment]
100% need this - being able to manually order or set a priority on items seems like a basic necessity, at least in the e-commerce space.

**Comment by codewizard13** [comment]
I'll add my "me too" - this is DEFINITELY a needed feature and seems so basic that it begs the question "Why isn't this in Elementor already?" Elementor is a great plugin and the dev team does awesome work. Would like to see this feature as soon as possible.

**Comment by nadworks** [comment]
Use case: Team members custom post type loop grid requires a custom order (either by custom field or backend custom post order). 
Seems Elementor often only think to the half-way point.

**Comment by portrayaloflife** [comment]
+1

**Comment by Kokouliss** [comment]
That would be very useful!

**Comment by awebguy99** [comment]
+1

**Comment by rf-999** [comment]
I am using ACF, Advanced Queries and WooCommerce, and the Elementor Pro Loop Grid.
If wanted to have a custom order query for my products. 
It took me nearly an hour to get it right because of a minor glitch, so I want to let you know about that.

If you want to have **custom order query** (and probably other Advanced Queries functionality) with WooCommerce Products:
* Loop Grid:
* * "Layout": Make sure you have selected **POSTS instead of PRODUCTS** (if you select Products, it won't work at all)
* * "Query": under "Source" you select PRODUCTS
* * now you can go on configuring with Advanced Queries

Hope this helps someone.

**Reply by ppwebox** [reply]
Thank you the contribution. I had this wrong, too, as I had selected the counter-intuitive "Layout" > "Products" option.

But what Elementor's Loop Grid widget really is still missing: the form field to let visitors select the sorting option (i.e. date, price, etc.). Because just of this I still have to rely on another plugin like FilterIt.

**Reply by DownHouse** [reply]
This feature was recently added. Previously, it was impossible to select a product.

**Comment by klitsios** [comment]
+1
I need it for custom sorting of Products.
Please fix this!

**Comment by fschtn** [comment]
Any news about this? :/

**Comment by AdamPoppleton** [comment]
+1 from me also ... sad that it's an issue that's been discussed here for so long and no response from the Elementor guys. Ho-hum ... an up-vote from me for this.

**Comment by portrayaloflife** [comment]
@nicholaszein any hope you can give us on this? Possible this is in the pipeline?

**Reply by nicholaszein** [reply]
@portrayaloflife,

I cannot comment on this request yet. 😉 🤫

**Comment by Studly** [comment]
I could really use this feature as well!

**Comment by Jacy-q-s** [comment]
Does anyone have a problem to query by current query recently?
It started to appeared "It's seems didn't found anything" frontend few days ago
I use Taxonomy as my query resource

**Comment by Mart-0ne** [comment]
Does Elementor team realizes that it makes their tool useless for any event based website ?
I've built hundreds of website with Elementor, and i'm beginning to get fed up of having to jump so many hoops for basic features, such as ordering by date an event list ...

**Comment by speedxcz** [comment]
+1 - I'm advocating for this functionality to be added. I can't even remember when I used the preset filtering types in Loop Grid. It's very limiting. Thanks.

**Comment by rendermouse** [comment]
How about also letting us sort by the order of Manual Selection? If a user specifically picks four individual posts, I would think most often they would also want specific control over exactly what order in which they appear.

**Comment by K-Gee** [comment]
I also need an urgent solution to order woocommerce product categories (displayed in an Elementor loop grid) by the date ACF field. Does anyone have any solutions? I have tried multiple php code snippets which all seem to hide categories altogether!

Thanks in advance!

**Reply by GwendleWeb** [reply]
I haven't tried this with woo commerce but with other acf fields in other post types my work around has been is the plugin 'advance post queries' by [Shabti Kaplan]. This gives you extra order by options in the loop query setting. Hope that helps!

**Reply by GwendleWeb** [reply]
Looks like there might be quirks with this and woocommerce , there are some comments higher in the chain that relate to products, not sure how it will work out in regards to categories sorry,  but I wish you the best of luck.

**Comment by K-Gee** [comment]
Thank you, I’ll try that.

Kind regards,
Kirstin

woodlandstudio.co.uk
07974715160



￼

**Comment by DevAndLetDie** [comment]
This would be amazing!

**Comment by Mart-0ne** [comment]
Does the elementor team care at all ? This issue has been pointed out TWO YEARS AGO
And they still roll out half-ass AI features while basic features still don't work.

**Comment by vgstef** [comment]
I really need this option. Any news about this? Is it in the roadmap? I'd like to have a follow up from Elementor team...
